### PR TITLE
Homework2

### PR DESCRIPTION
**Pull request by mattsloe**
homework 2 completed

### CODE REVIEW

**Comment by dstad418** [comment]
Looks good, I *really* like your comments in the code, even if some find it superfluous. I noted some of your id's like "anythingElse" that were used, it's something I did too, but someone commented that it could be more explanatory, but that may be splitting hairs.
All-in-all, really solid stuff. Great job, Matt!

**Review by SayleeMJ — COMMENTED** [review]
The only thing I want to point out is that the color of the highlighted text is a bit light. To make it more visible, you might want to try using some darker shades. Overall though, I think your code looks great and it's clean!

File: hw2/03-count.js [hunk 1]
```diff
@@ -1,3 +1,18 @@
...
+  // retrieve search term from input
+  const search = event.target.value.trim();
+  if (search === '') {
+    content.innerHTML = content.textContent;
+    return;
+  }
+  // retrieve the content to be searched
+  const { textContent } = content;
+  // find and highlight all occurrences of the search term in the text content
+  const highlightedTxt = textContent.replace(new RegExp(search, 'gi'), (match) => `<mark>${match}</mark>`);
```
Comment (lebenpdx) [inline hw2/03-count.js]: We don't want to highlight matches that are inside other words, just whole word matches. You can do this by adding word boundary anchors(`\b`), around the input, to your regex.

File: hw2/05-form.js [hunk 2]
```diff
@@ -1 +1,29 @@
...
+  const anythingElse = document.getElementById('anythingElse').value;
+
+  // print to console
+  console.log({
+    fullName,
+    email,
+    registrationStatus,
+    selectedCourses,
+    anythingElse,
+  });
```
Comment (lebenpdx) [inline hw2/05-form.js:28]: I would just format this, so that it is a little bit easier to read.

**Review by lebenpdx — COMMENTED** [review]
I have finished my review

**Pull request merged**